hazelcast/src/main/java/com/hazelcast/internal/config/ConfigValidator.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.config;
+
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.NearCacheConfig;
+import com.hazelcast.instance.BuildInfoProvider;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import static com.hazelcast.config.InMemoryFormat.NATIVE;
+import static com.hazelcast.config.MapConfig.DEFAULT_EVICTION_PERCENTAGE;
+import static com.hazelcast.config.MapConfig.DEFAULT_MIN_EVICTION_CHECK_MILLIS;
+
+/**
+ * Validates a Hazelcast configuration in a specific context like OS vs. EE or client vs. member nodes.
+ */
+public final class ConfigValidator {
+
+    private static final boolean IS_ENTERPRISE = BuildInfoProvider.getBuildInfo().isEnterprise();
```

Comment:
I had to add this check for Enterprise, since the `AbstractClientInternalCacheProxy` is also used in Hazelcast Enterprise, which of course works fine with `NATIVE`.